test/test_cuda.py

```diff
@@ -322,6 +322,7 @@ def test_set_per_process_memory_fraction(self):
         # ensure out of memory error doesn't disturb subsequent kernel
         tensor.fill_(1)
         self.assertTrue((tensor == 1).all())
+        torch.cuda.set_per_process_memory_fraction(1.0, 0)
```

Comment:
We should create a context manager / fixture for this going forward...

Reply:
Yeah this is pretty annoying/involved atm because it requires e.g., changing the PluggableAllocator interface